src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindIntegrationTest.java

```diff
@@ -1269,6 +1270,134 @@ public void byDateColumn() {
           .body("data.documents[0]", jsonEquals(expected))
           .body("data.documents", hasSize(1));
     }
+
+    @Test
+    public void simpleOr() {
+      String json =
+          """
+                {
+                    "find": {
+                        "filter": {
+                            "$or": [
+                                {"username" : "user1"},
+                                {"username" : "user2"}
+                            ]
+                        }
+                    }
+                }
+
+              """;
+
+      String expected1 =
+          """
+                  {"_id":"doc1", "username":"user1", "active_user":true, "date" : {"$date": 1672531200000}}
+                  """;
+      String expected2 =
+          """
+                  {"_id":"doc2", "username":"user2", "subdoc":{"id":"abc"},"array":["value1"]}
+                  """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("status", is(nullValue()))
+          .body("errors", is(nullValue()))
+          .body("data.documents", hasSize(2));
+    }
+
+    @Test
+    public void nestedAndOr() {
+      String json =
+          """
+{
+    "find": {
+        "filter": {
+            "$or": [
```

Comment:
I'd recommend making this test use a `$and` of `$or`, rather than a `$or` of `$and`. That's the most common use case for `$and`, because there's no difference between `$and: [{ username: 'user3' }, { 'subdoc.id': 'abc' }]` and `{ username: 'user3', 'subdoc.id': 'abc' }`. You would typically use `$and` like this:

```javascript
{
  $and: [
    { $or: [{ username: 'user1' }, { 'subdoc.id': 'test1' }] },
    { $or: [{ username: 'user2' }, { 'subdoc.id': 'test2' }] }
  ]
}
```

Reply:
thank, Val. Fixed